moped-editor/src/views/projects/projectView/ProjectComponents/ComponentForm.js

```diff
@@ -164,6 +169,13 @@ const ComponentForm = ({
     disable: isEditingExistingComponent,
   });
 
+  useResetDependentFieldOnSignalChange({
+    parentValue: watch("signal"),
+    dependentFieldName: "locationDescription",
+    valueToSet: signal ? getSignalOptionLabel(signal) : "",
```

Comment:
i worry that we might nullify information that a user intended to keep in this field. E.g., they pick a signal component, edit the location description, then change the component to something else.

to be safe, what if we also `watch` the `location_description` and use that in `valueToSet` instead of `""` (if it's not null)? does that make sense? but that might make the hook go loopy 🤷